src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -166,6 +169,10 @@ private async ValueTask<Response> GetResponseAsync(bool async, string uri, Cance
         private HttpMessage CreateRequest(string uri)
         {
             HttpMessage message = _pipeline.CreateMessage();
+            if (!string.IsNullOrEmpty(_userAgent))
+            {
+                message.SetProperty("UserAgentOverride", _userAgent!);
```

Comment:
I think we can just set the header here and not use the override mechanism?

request.Headers.SetValue

Reply:
Because we have the `MgmtTelemetryPolicy`, which will re-write the `User-Agent` value, so this command `request.Headers.SetValue`will not work.